src/activity_actor.cpp

```diff
@@ -2139,13 +2140,77 @@ std::unique_ptr<activity_actor> stash_activity_actor::deserialize( JsonIn &jsin
     return actor.clone();
 }
 
+void bookbinder_copy_activity_actor::start( player_activity &act, Character & )
+{
+    act.moves_total = to_moves<int>( pages * 2_minutes );
+    act.moves_left = to_moves<int>( pages * 2_minutes );
+}
+
+void bookbinder_copy_activity_actor::finish( player_activity &act, Character &p )
+{
+    item *it_bkbinder = book_binder.get_item();
+
+    const auto old_recipes = it_bkbinder->get_var( "EIPC_RECIPES" );
+    if( old_recipes.empty() ) {
+        it_bkbinder->set_var( "EIPC_RECIPES", "," + r_id + "," );
```

Comment:
Forgive me for being unfamiliar with this code, but it's supposed to look like `,recipe1,,recipe2,,recipe3,`?

Reply:
I think it's just as you described.
I copied it from the way the `einkpc_download_memory_card` function stores its recipes. I didn't change the way it was stored to avoid any problems when `item::get_available_recipes` loads the recipe.

Reply:
That seems odd, but nothing for you to deal with. You could consider adding a function or two to `item` to move the business logic of storing and checking the recipes elsewhere to make this a little cleaner and not duplicate code.